src/ImageFormatStore.php

```diff
@@ -1,63 +0,0 @@
-<?php
-
-namespace TractorCow\SilverStripeImageFormatter;
-
-use League\Flysystem\Filesystem;
-use SilverStripe\Assets\FilenameParsing\FileResolutionStrategy;
-use SilverStripe\Assets\FilenameParsing\ParsedFileID;
-use SilverStripe\Assets\Flysystem\FlysystemAssetStore;
-
-class ImageFormatStore extends FlysystemAssetStore
-{
-    /**
-     * Stop caring about hash when checking for file collisions; This breaks many things.
-     * Specifically, the "original" image and the formatted image will have different hashes,
-     * but we cannot determine the formatted hash without generating it, or storing it somewhere.
-     *
-     * @param string $filename
-     * @param string $hash
-     * @param null $variant
-     * @return bool|mixed
-     */
-    public function exists($filename, $hash, $variant = null)
-    {
-        if (empty($filename) || empty($hash)) {
-            return false;
-        }
-
-        $parsedFileID = new ParsedFileID($filename, $hash, $variant);
-
-        $publicSet = [
-            $this->getPublicFilesystem(),
-            $this->getPublicResolutionStrategy(),
-        ];
-
-        $protectedSet = [
-            $this->getProtectedFilesystem(),
-            $this->getProtectedResolutionStrategy(),
-        ];
-
-        /** @var Filesystem $fs */
-        /** @var FileResolutionStrategy $strategy */
-
-        // First we try to search for exact file id string match
-        foreach ([$publicSet, $protectedSet] as $set) {
-            list($fs, $strategy) = $set;
-            $fileID = $strategy->buildFileID($parsedFileID);
-            if ($fs->has($fileID)) {
```

Comment:
@xini has ss 5.2 fixed the issue where a file with a different hash was treated as non-existant, even though the path matched? The exists() used to do a hash check and fail if it was false, causing a false negative when the file actually existed.

Reply:
I assume that manipulateExtension() does the check of the original hash vs the original file (old extension) which is something that didnt work in ss4.

Reply:
@tractorcow yes, SS 5.2 officially allows file variants with different extensions and also recognises them when they already exist.